src/fides/api/alembic/migrations/versions/5f96f13be8ec_property_specific_messaging_migration_.py

```diff
@@ -0,0 +1,187 @@
+        insert_into_property_query: TextClause = text(
+            "INSERT INTO plus_property (id, name, is_default, type)"
+            "VALUES (:id, :name, :is_default, :type)"
+        )
+        characters = string.ascii_uppercase + string.digits
+        new_property_id: str = "FDS-" + "".join(random.choices(characters, k=6))
+        default_property_id = new_property_id
+
+        new_property: Dict[str, Any] = {
+            "id": new_property_id,
+            "name": "Default Property Autogenerated",
+            "is_default": True,
+            "type": "website",
+        }
+        try:
+            bind.execute(insert_into_property_query, new_property)
+        except IntegrityError as exc:
+            raise Exception(
+                f"Fides attempted to create a new default property but got error: {exc}. "
+            )
+
+    # STEP 2: Ensure all default templates are saved to the DB
+    existing_templates: LegacyCursorResult = bind.execute(
+        text("SELECT id, type from messaging_template;")
+    )
+    templates_from_db: List[str] = [
+        template.type for template in existing_templates.all()
+    ]
+    for template_type, template in DEFAULT_MESSAGING_TEMPLATES.items():
+        # if a template is not already in the DB, save it with defaults
+        if template_type not in templates_from_db:
+            insert_into_messaging_template_query: TextClause = text(
+                "INSERT INTO messaging_template (id, type, content, is_enabled)"
+                "VALUES (:id, :type, :content, :is_enabled)"
+            )
+            new_messaging_template: Dict[str, Any] = {
+                "id": generate_record_id("mes"),
+                "type": template_type,
+                "content": template["content"],
+                "is_enabled": False,
+            }
+            try:
+                bind.execute(
+                    insert_into_messaging_template_query.bindparams(
+                        bindparam("content", type_=JSONB)
+                    ),
+                    new_messaging_template,
+                )
+            except IntegrityError as exc:
+                raise Exception(
+                    f"Fides attempted to create a new messaging_template but got error: {exc}. "
+                )
+
+    # STEP 3: Ensure all saved messaging templates are linked to the default property
+    updated_templates: ResultProxy = bind.execute(
+        text("select id, type from messaging_template;")
+    )
+    for template in updated_templates:
+        insert_into_messaging_template_to_property_query: TextClause = text(
+            "INSERT INTO messaging_template_to_property (id, messaging_template_id, property_id)"
+            "VALUES (:id, :messaging_template_id, :property_id)"
+        )
+        new_messaging_template_to_property: Dict[str, Any] = {
+            "id": generate_record_id("mes"),
+            "messaging_template_id": template["id"],
+            "property_id": default_property_id,
```

Comment:
You don't define `default_property_id` now in the case where there was already exactly one property, which would cause a failure on this line:    ` "property_id": default_property_id,`

Reply:
Ah you're right, thanks!